lec15-data-viz.Rmd

```diff
@@ -1,4 +1,341 @@
+  theme(plot.title = element_text(hjust = 0.5), 
+        text = element_text(size = 8, family='Times'),
+        legend.title=element_blank())
+  
+```
+
+
+You can add your own alterations to themes. For example, here is an APA theme built on top of the base theme_bw(). Themes can be saved into variables so they don't need to be typed out every time you use a graph.
+
+
+```{r}
+
+apatheme <- theme_bw()+
+  theme(panel.grid.major=element_blank(),
+        panel.grid.minor=element_blank(),
+        panel.border=element_blank(),
+        axis.line=element_line(),
+        text=element_text(family='Times'),
+        legend.title=element_blank())
+
+ggplot(iris, aes(x=Sepal.Width, y=Sepal.Length, color=Species)) + geom_point()+
+  ylab("Sepal Length") +
+  xlab ("Sepal Width") +
+  ggtitle("Sepal Width by Sepal Length") +
+  apatheme
+  
+```
+
+
+## Bar graphs
+
+What if we want a bar graph of Sepal Length by Species?
+```{r}
+ggplot(iris, aes(x=Species, y=Sepal.Length)) + geom_bar(stat="identity")+
+  ylab("Sepal Length") +
+  xlab ("Species") +
+ggtitle("Sepal Width by Sepal Length") +
+  theme_bw() +
+  theme(plot.title = element_text(hjust = 0.5))
+```
+
+## Violin plots
+
+```{r}
+ggplot(iris, aes(x=Species, y=Sepal.Length)) + geom_violin()+
+  ylab("Sepal Length") +
+  xlab ("Species") +
+ggtitle("Sepal Width by Sepal Length") +
+  theme_bw() +
+  theme(plot.title = element_text(hjust = 0.5))
+```
+
+## Box plots
+
+
+ We can use boxplots to visualize the distribution Sepal Length within each species:
+
+ ```{r boxplot}
+ ggplot(data = iris, aes(x = Species, y = Sepal.Length)) +
+     geom_boxplot()
+ ```
+
+> #### Challenge
+>
+> Would you say this is more or less informative than the violin plot we just made?
+>
+
+By adding points to boxplot, we can have a better idea of the number of
+measurements and of their distribution:
+
+ ```{r boxplot-with-points}
+ ggplot(data = iris, aes(x = Species, y = Sepal.Length)) +
+     geom_boxplot(alpha = 0) +
+     geom_jitter(alpha = 0.3, color = "tomato")
+ ```
+ 
+ Notice how the boxplot layer is behind the jitter layer? What do you need to
+ change in the code to put the boxplot in front of the points such that it's not
+ hidden?
+ 
+> #### Challenge
+>
+> So far, we've looked at the distribution of sepal length within species.  Try making a
+> new plot to explore the distribution of another variable within each species.
+>
+
+
+What about making interactive graphs in R?
+There is a package called plotly that lets us make our ggplot2 graphs interactive:
+```{r}
+library(plotly)
+g <- ggplot(iris, aes(Species)) + geom_bar()
+p=ggplotly(g)
+p
+```
+
+> #### Challenge
+>
+> Create another type of interactive graph (i.e. not a bar graph) using these data. What kinds of graphs
+> might interactivity add or improve the communication and clarity of your message to?
+>
+
+Here is qplot, a way of making super quick scatterplots in R.
+```{r}
+qplot(Petal.Length, Sepal.Length, data = iris)
```

Comment:
They've already covered `qplot` mostly (though there is much more you could talk about with `qplot`)

Reply:
i might save it for next lecture then if I can think of a valuable way of covering it